src/language/rule/rule.ts

```diff
@@ -18,6 +18,53 @@
 import * as ts from "typescript";
 import {RuleWalker} from "../walker/ruleWalker";
 
+export interface IRuleMetadata {
+   /**
+    * The kebab-case name of the rule.
+    */
+    ruleName: string;
+
+   /**
+    * The type of the rule - its overall purpose
+    */
+    type: RuleType;
+
+   /**
+    * A short, one line description of what the rule does.
+    */
+    description: string;
+
+   /**
+    * More elaborate details about the rule.
+    */
+    descriptionDetails?: string;
+
+   /**
+    * An explanation of the available options for the rule.
+    */
+    optionsDescription?: string;
+
+   /**
+    * Schema of the options the rule accepts.
+    * The first boolean for whether the rule is enabled or not is already implied.
+    * This field describes the options after that boolean.
+    * If null, this rule has no options and is not configurable.
```

Comment:
jsyaml can handle `null`, but not `undefined`, so I was actually meant `null` specifically. Maybe it would be better to just make it optional and handle things in the docs build script, but I also don't mind requiring an explicit null